pyanaconda/payload/install_tree_metadata.py

```diff
@@ -0,0 +1,244 @@
+# Gather and provides metadata about installation tree.
+#
+# Copyright (C) 2018  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+import time
+import requests
+import os
+
+from productmd.treeinfo import TreeInfo
+from pyanaconda.core import util, constants
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+MAX_TREEINFO_DOWNLOAD_RETRIES = 6
+
+
+class InstallTreeMetadata(object):
+
+    def __init__(self):
+        self._tree_info = TreeInfo()
+        self._meta_repos = []
+        self._session = util.requests_session()
```

Comment:
It looks like this could be created locally in `load_url`.

Reply:
Fixed